lib/setup.js

```diff
@@ -85,56 +86,11 @@ PouchDB.destroy = utils.toPromise(function (name, opts, callback) {
     name = undefined;
   }
 
-  if (!opts.internal) {
-    console.log('PouchDB.destroy() is deprecated and will be removed. ' +
-                'Please use db.destroy() instead.');
-  }
-
-  var backend = PouchDB.parseAdapter(opts.name || name, opts);
-  var dbName = backend.name;
-  var adapter = PouchDB.adapters[backend.adapter];
-  var usePrefix = 'use_prefix' in adapter ? adapter.use_prefix : true;
-  var baseName = usePrefix ?
-    dbName.replace(new RegExp('^' + PouchDB.prefix), '') : dbName;
-  var fullName = (backend.adapter === 'http' || backend.adapter === 'https' ?
-      '' : (opts.prefix || '')) + dbName;
-  function destroyDb() {
-    // call destroy method of the particular adaptor
-    adapter.destroy(fullName, opts, function (err, resp) {
-      if (err) {
-        callback(err);
-      } else {
-        PouchDB.emit('destroyed', name);
-        //so we don't have to sift through all dbnames
-        PouchDB.emit(name, 'destroyed');
-        callback(null, resp || { 'ok': true });
-      }
-    });
-  }
-
-  var createOpts = utils.extend(true, {}, opts, {adapter : backend.adapter});
-  new PouchDB(baseName, createOpts, function (err, db) {
+  new PouchDB(name, opts, function (err, db) {
```

Comment:
We have already had a deprecation notice for quite a while, how about we just remove it now and next version is a major version bump?